### PR TITLE
Added File Handling Tools and Environment Details

### PR DESCRIPTION
**Pull request by jahnavib26**
### **New Tool Implementations:**

1. **`read_file` Tool**:
   - Reads and retrieves content from a file specified by the user.
   - Handles edge cases such as invalid file paths and special characters.

2. **`list_files` Tool**:
   - Lists files in a specified directory.
   - Supports recursive listing of subdirectories.
   - Includes robust error handling for cases like missing directories.

3. **`search_files` Tool**:
   - Searches for a user-defined regular expression within files in a directory.
   - Allows filtering by file extensions and supports case-insensitive searches.
   - Limits results to a configurable maximum for improved performance.

### **Environment Details:**
- Improved the **environment details context** for the assistant to include:
  - A list of visible files in the current VSCode workspace.
  - Open tabs across all VSCode tab groups.
- Provides better context for task-related interactions and debugging.

### CODE REVIEW

File: src/core/index.ts [hunk 1]
```diff
@@ -130,14 +131,24 @@ export class DevAssist {
...
 					if (this.assistantMessageContent[this.assistantMessageContent.length - 1].type === "tool_use" && this.assistantMessageContent[this.assistantMessageContent.length - 1].name === "ask_followup_question") {
 						this.askFollowup = true;
 						this.askFollowupIndex = this.assistantMessageContent.length - 1;
 					}
-					// console.log("assistantMessageContent", this.assistantMessageContent);
+					if (this.assistantMessageContent[this.assistantMessageContent.length - 1].partial === true && this.assistantMessageContent[this.assistantMessageContent.length - 1].type === "text" && this.assistantMessageContent[this.assistantMessageContent.length - 1].content === '') {
+					// remove that element from the array becaus it is partial content
+					this.assistantMessageContent.pop();
+					}
+
```
Comment (adithya1012) [inline src/core/index.ts:145]: As we implemented this.currentStreamingContentIndex removing the empty text object is not necessary. I will update this code in next PR.

**Review by adithya1012 — APPROVED** [review]
Looks Good !

**Review by hdprajwal — APPROVED** [review]
Looks good to me. I will integrarte it my code for Tools use in API Provider and raise a PR.

**Review by Sinduja98 — APPROVED** [review]
Looks good for me.

**Review by nidhitalastha — APPROVED** [review]
LGTM!

**Pull request merged**